lighthouse.go

```diff
@@ -869,47 +868,76 @@ func (lh *LightHouse) SendUpdate() {
 		},
 	}
 
-	if v == 1 {
-		var relays []uint32
-		for _, r := range lh.GetRelaysForMe() {
-			if !r.Is4() {
-				continue
-			}
-			b := r.As4()
-			relays = append(relays, binary.BigEndian.Uint32(b[:]))
-		}
+	nb := make([]byte, 12, 12)
+	out := make([]byte, mtu)
 
-		msg.Details.OldRelayVpnAddrs = relays
-		//TODO: assert ipv4
-		b := lh.myVpnNetworks[0].Addr().As4()
-		msg.Details.OldVpnAddr = binary.BigEndian.Uint32(b[:])
+	var v1Update, v2Update []byte
+	var err error
+	var v cert.Version
+	updated := 0
+	lighthouses := lh.GetLighthouses()
 
-	} else if v == 2 {
-		var relays []*Addr
-		for _, r := range lh.GetRelaysForMe() {
-			relays = append(relays, netAddrToProtoAddr(r))
+	for lhVpnAddr := range lighthouses {
+		hi := lh.ifce.GetHostInfo(lhVpnAddr)
+		if hi != nil {
+			v = hi.ConnectionState.myCert.Version()
+		} else {
+			v = lh.ifce.GetCertState().defaultVersion
 		}
-		msg.Details.RelayVpnAddrs = relays
-		msg.Details.VpnAddr = netAddrToProtoAddr(lh.myVpnNetworks[0].Addr())
+		if v == cert.Version1 {
```

Comment:
Since we are reusing the `msg` struct we need to zero out the v2 fields in v1 and v1 fields in v2.